Doc/reference/compound_stmts.rst

```diff
@@ -796,12 +796,12 @@ The :keyword:`!async for` statement
 .. productionlist:: python-grammar
    async_for_stmt: "async" `for_stmt`
 
-An :term:`asynchronous iterable` is able to call asynchronous code in its
-*iter* implementation, and :term:`asynchronous iterator` can call asynchronous
-code in its *next* method.
+An :term:`asynchronous iterable` provides an *iter* method that returns an
```

Comment:
Technically `*iter*` is `__aiter__` and `*next*` is `__anext__`.

I have no idea how to write the text better.  Maybe it should not name iter/next at all but define async-iterable/async-iterator in plain English words?

Reply:
I agree it's a bit unclear, but consistent with how async context managers are described in the `async with` section.

Reply:
I believe `async for` description should be compatible with `for`, `async with` should follow `with` structure.

@1st1 @aeros please share your opinions.

Reply:
Beware english not my primary language, but here's my proposed wording:

    An :term:`asynchronous iterable` provides an ``__aiter__`` method that directly returns an
    :term:`asynchronous iterator`, which can call asynchronous code in its ``async __anext__`` method.

Reply:
Looks good.

@1st1 @aeros maybe you can review the proposal?

Reply:
I like Julien's phrasing.

Reply:
+1 from me to go with Julian's proposal.

Reply:
I'm not too keen on this change as I think the reason for using italicized words originally was to refer to the methods without getting into the specifics which are described in more detail in the data model documentation for asynchronous iterators.  However it seems to have majority support so I updated the PR accordingly.

Reply:
The reference section of the docs is typically intended to be more technically accurate, rather than prioritizing a simplified introduction. I wouldn't be opposed to applying the same to `async with` (to instead refer to `__aexit__` and `__aenter__`), but don't think it's worth changing by itself.